resolwe_bio/processes/differential_expression/cuffdiff.yml

```diff
@@ -148,8 +148,9 @@
 
       {% for c in control %}
         {% for t in case %}
-          {% if control.0.source != c.source or c.source != t.source %}
-            re-error "DE analysis must be crated using the expression data from the same source."
+          {% if (control|first).source != c.source or c.source != t.source %}
+            re-warning "Genes in all expression data must be annotated by the same genome database."
+            re-error "Sample {{ c|name }} has {{ c.source }} gene IDs, while {{ t|name }} has {{ t.source }}."
```

Comment:
Incorrect error message. The `if` statement checks for two things: (1) the source of the current control sample must be the same as the source of the first control sample, which is not handled in the error message, and (2) the source of the current control sample must be the same as the source of the current case sample.

Reply:
@mzganec I understand the if statement, however even if the first control sample will have wrong source, error message will list it. I don't think it make sense to make Error message more complicated.

Reply:
OK.